File: Difficulty: Easy/Day before N days/day-before-n-days.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

// Function to find the nth day relative to a given day
int nthDay(int d, int n) {
    // Calculate the nth day by subtracting n modulo 7 from d and ensuring the result is within the range [0,6]
    return (d - n % 7 + 7) % 7;
}



//{ Driver Code Starts.

int main() {

    int t;
    cin >> t;
    while (t-- > 0) {
        int d, n;
        cin >> d >> n;
        cout << nthDay(d, n) << endl;
    }
}
// } Driver Code Ends

File: Difficulty: Easy/Day before N days/day-before-n-days.py
#{ 
 # Driver Code Starts
#Initial Template for Python 3

# } Driver Code Ends

# User function Template for python3

# Read inputs
d = int(input())  # Current day index (0=Monday, 1=Tuesday, ..., 6=Sunday)
n = int(input())  # Number of days before the current day

# Calculate the day index `n` days before day `d`
previous_day = (d - n) % 7

# If result is negative, wrap around the week
if previous_day < 0:
    previous_day += 7

# Print the result
print(previous_day)


#{ 
 # Driver Code Starts.

# } Driver Code Ends